### PR TITLE
NN-3145 filter cleared holdings out of the list

### PR DESCRIPTION
**Pull request by JamieDraperUK**

### CODE REVIEW

File: views/prisonerProfile/prisonerFinance/privateCash.njk [hunk 1]
```diff
@@ -33,8 +33,7 @@
     {{ govukTable({
       head: [
         { text: "Date" },
-        { text: "Money in" },
-        { text: "Money out" },
+        { text: "Amount" },
```
Comment (petergphillips) [inline views/prisonerProfile/prisonerFinance/privateCash.njk:36]: Isn't this reliant on `mergeAmounts` being `true`?

File: backend/shared/createTransactionViewModel.js [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
   transactions.map(transaction => {
     const { description: prisonName } = prisons.find(agency => transaction.agencyId === agency.agencyId)
     const { penceAmount, currentBalance } = transaction
 
     return [
       { text: transaction.entryDate && formatTimestampToDate(transaction.entryDate) },
-      { text: transaction.postingType === 'CR' ? formatCurrency(penceAmount / 100) : '' },
-      { text: transaction.postingType === 'DR' ? formatCurrency(penceAmount / 100) : '' },
+      ...(mergeAmounts
+        ? [{ text: (penceAmount && formatCurrency(penceAmount / 100)) || '' }]
```
Comment (petergphillips) [inline backend/shared/createTransactionViewModel.js:11]: I would have thought that `formatCurrency` should be able to cope with `penceAmount` being `0` or `false`?  Is it that if the value is `0` you don't want to display anything

**Review by petergphillips — APPROVED** [review]
LGTM

File: views/prisonerProfile/prisonerFinance/privateCash.njk (hunk 1 above)
Comment (JamieDraperUK) [reply views/prisonerProfile/prisonerFinance/privateCash.njk:36]: It's always true for that section. Having a central mapper that works for difference screens is a bit of a smell here and is causing confusion.

File: backend/shared/createTransactionViewModel.js (hunk 2 above)
Comment (JamieDraperUK) [reply backend/shared/createTransactionViewModel.js:11]: Agreed. Will improve it in the next PR.

**Pull request merged**